ocis/pkg/runtime/service/service.go

```diff
@@ -64,6 +64,8 @@ import (
 var (
 	// runset keeps track of which services to start supervised.
 	runset map[string]struct{}
+	// time to wait after starting the preliminary services
+	_preliminaryDelay = 6 * time.Second
```

Comment:
This is actually a good point, but I don't know how to avoid that for the moment.

In a perfect world they would assembly gracefully. But there is no such mechanism implemented for the registry (neither for the cash). If services can't connect to the registry they die immediately. Additionally, if services die during initialization of the single binary, they will not come up again (at least  this what I observed in various runs). 

Is the single binary supposed to restart services during initialization? Wouldn't that lead to infinite restarts when the services are misconfigured?

> Why is our built in supervisor so different than a kubernetes cluster?

This question I don't understand. How could our selfmade supervisor be like a kubernetes cluster? Is it supposed to be?